recommender/engines/base/base_neural_network.py

```diff
@@ -0,0 +1,61 @@
+# pylint: disable=too-few-public-methods, line-too-long
+"""Abstract neural network"""
+
+from abc import ABC
+from typing import Tuple
+from itertools import chain
+
+from torch.nn import Linear, BatchNorm1d, ReLU
+
+
+class BaseNeuralNetwork(ABC):
+    """Abstract Neural Network"""
+
+    @staticmethod
+    def _create_layers(
+        input_dim: int,
+        output_dim: int,
+        layers_sizes: Tuple[int],
+        inc_batchnorm: bool = True,
+        activation=ReLU,
+    ):
+        """
+        Create neural network layers based on a provided tuple which contains the size of layers
+
+        Args:
+            input_dim: input dimensionality,
+            output_dim: output dimensionality,
+            layers_sizes: a tuple of target layers sizes,
+            inc_batchnorm: include BatchNorm1d?
+            activation: activation function
+        """
+        # The first layer
+        layers = [
+            Linear(input_dim, layers_sizes[0]),
+            BatchNorm1d(layers_sizes[0]),
+            activation(),
+        ]
+        if not inc_batchnorm:
+            del layers[1]  # delete batchnorm
+
+        # The middle layers
+        layers += list(
+            chain.from_iterable(
+                [
+                    [
+                        Linear(n_size, next_n_size),
+                        BatchNorm1d(next_n_size),
+                        activation(),
+                    ]
+                    if inc_batchnorm
+                    else [Linear(n_size, next_n_size), activation()]
```

Comment:
This could be extracted to a private function `_create_layer` and then used here and also in the lines 33-39

Reply:
More clarity and DRY code